pkg/api/security/security.go

```diff
@@ -256,3 +256,21 @@ func saveAuthToken(token, tokenPath string) error {
 	log.Infof("Wrote auth token in %s", tokenPath)
 	return nil
 }
+
+// call the file.Sync method to flush the file to disk and catch potential I/O errors
+func writeFileAndSync(name string, data []byte, perm os.FileMode) error {
+	f, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(data)
+	// only sync data if the write was successful
+	if err == nil {
+		err = f.Sync()
+	}
+	// but always close the file and report the first error that occurred
+	if err1 := f.Close(); err1 != nil && err == nil {
+		err = err1
+	}
+	return err
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Using `defer` in this kind of scenario is a better idiomatic‏ go IMO.
```suggestion
	f, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)
	if err != nil {
		return err
	}
	defer f.Close()
	
	_, err = f.Write(data)
	if err != nil {
		return err
	}
	
	err = f.Sync()
	if err != nil {
		return err
	}
	
	return nil
```